scalableminds/webknossos
# Make Project and TaskType names unique by Organization

Project and TaskType names had to be globally unique. Now, they are uniqe per organization.

### TODO
 - [x] change unique constraints in schema
 - [x] add organization during insert
 - [x] change project api (by orga+name/by id/by disambiguation)?
 - [x] change edit view route
 - [x] use new api in frontend
 - [x] adapt task search form
 - [x] task search form persistence? → follow-up https://github.com/scalableminds/webknossos/issues/5337
 - [x] how to access project name for rendering form?
 - [x] adapt access queries
 - [x] rest api versioning
 - [x] snapshot tests

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- List projects, create project, edit project, should all still work as expected
- Create tasks, search for tasks by project, should show.
- Try to create second project with the same name, should fail with readable message.
- Create second organization
- Create project with name used in other org, should work
- Edit projects, the ones across organizations should not influence each other
- same with task types.

### Issues:
- fixes #4005

------
